exercises/binary-search/src/example/java/BinarySearch.java

```diff
@@ -0,0 +1,40 @@
+
+import java.util.List;
+
+public class BinarySearch<T extends Comparable<T>> {
```

Comment:
Personally, I currently give *wide berth* to implementors in what happens in the example.  As long as it passes the tests and represents a true solution, it's generally been implementor's choice.

I think this way because practitioners don't see these.  The high value conversations in PRs are about how the test suite flows, the API decisions therein and whether we're implementing the exercise correctly.